src/riak_kv_vnode.erl

```diff
@@ -2383,12 +2383,29 @@ encode_and_put_no_sib_check(Obj, Mod, Bucket, Key, IndexSpecs, ModState,
                         false ->
                             ok
                     end,
-                    PutRet = Mod:put(Bucket, Key, IndexSpecs, EncodedVal,
-                                     ModState),
+                    PutRet = case timeseries_data(Key, Obj) of
+                                 true ->
+                                     RawKey = sext:decode(Key),
+                                     Mod:put(Bucket, RawKey, IndexSpecs, EncodedVal,
+                                             ModState);
+                                 false ->
+                                     Mod:put(Bucket, Key, IndexSpecs, EncodedVal,
+                                             ModState)
+                             end,
                     {PutRet, EncodedVal}
             end
     end.
 
+%% Look for sext-encoded tuple (key) because handoff lifts it straight out of
+%% the backend and gives it to us as is. If this isn't sext-encoded
+%% then this ain't timeseries so we don't need to dig through the
+%% object metadata for a DDL.
+timeseries_data(<<16,0,0,0, _Rest/binary>>, Object) ->
+    {IsTs, _Ver} = riak_object:is_ts(Object),
```

Comment:
I'll clarify the comment. The point is that we *do* check for a DDL via `is_ts/1` but we only do that if the key looks like it's a sext-encoded tuple.

Does that alleviate your concern? I don't think any code changes are warranted, simply improving the commentary.